### DISCUSSION TITLE
`<Script />` from `next/script` with `strategy="beforeInteractive"` does not work with `crossOrigin` jsx attribute

### DISCUSSION THREAD

**Discussion by nithiwatter** [vercel/next.js#43518] [Help]
### What version of Next.js are you using?

11.0.1 (but also tested with next@canary)

### What version of Node.js are you using?

14.17.3

### What browser are you using?

Edge/Chrome

### What operating system are you using?

Windows

### How are you deploying your application?

next run dev

### Describe the Bug

Hi everyone,

Please help me with the correct way to use next/script component. I am trying to add crossorigin attribute to the <script> tag via the following code

`<Script src="https://code.jquery.com/jquery-3.5.1.slim.min.js" integrity="sha384-DfXdz2htPH0lsSSs5nCTpuj/zy4C+OGpamoFVy38MVBnE+IbbVYUew+OrCXaRkfj" crossOrigin="anonymous" strategy="beforeInteractive" />`

However, it is not working. When I view page source, no attribute called crossorigin is added to my script. I am not sure if this is a bug or not. It seems like the attribute would only occur if I do this:

`<Script src="https://code.jquery.com/jquery-3.5.1.slim.min.js" integrity="sha384-DfXdz2htPH0lsSSs5nCTpuj/zy4C+OGpamoFVy38MVBnE+IbbVYUew+OrCXaRkfj" crossorigin="anonymous" strategy="beforeInteractive" />`

(lowercasing `crossOrigin` to `crossorigin`), but in that case, I get a warning that that is not a valid jsx attribute instead. 

### Expected Behavior

Expect to have `crossorigin` added to `<script>` when viewing page source/markup. 

### To Reproduce

Try appending (using the new `<Script>` component from `next/script`)

`<Script
                src="https://code.jquery.com/jquery-3.5.1.slim.min.js"
                integrity="sha384-DfXdz2htPH0lsSSs5nCTpuj/zy4C+OGpamoFVy38MVBnE+IbbVYUew+OrCXaRkfj"
                crossOrigin="anonymous"
                strategy="beforeInteractive"
            />`
`            <Script
                src="https://cdn.jsdelivr.net/npm/bootstrap@4.6.0/dist/js/bootstrap.bundle.min.js"
                integrity="sha384-Piv4xVNRyMGpqkS2by6br4gNJ7DXjqk09RmUpJ8jgGtD7zP9yug3goQfGII0yAns"
                crossorigin="anonymous"
                strategy="beforeInteractive"
            />`

in `_app.js`: Notice that the first tag uses `crossOrigin` and the second does not. You should get a warning that bootstrap Javascript could not run as it requires jquery (the first jquery script would fail to load as it lacks the `crossorigin` attribute in the page markup).

**Comment by curly210102** [comment]
`afterInteractive` and `lazyOnload` script is been injected in client-side. `beforeInteractive` script is rendered in server-side. There are different code logic, `afterInteractive` and `lazyOnload` works with `crossOrigin` attribute, but `beforeInteractive` have to work with `crossOrigin` in next.config.js (https://nextjs.org/docs/messages/doc-crossorigin-deprecated). @nithiwatter you can try to config the next.config.js and restart the dev server.

**Comment by nithiwatter** [comment]
Thank you so much for clarifying. Will this be fixed in upcoming versions or should I just config the env variable as the final solution.
